### PR TITLE
Update to 11.8, add pyclibrary requirements, reset the build number

### PR DESCRIPTION
**Pull request by m3vaz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by m3vaz** [comment]
@conda-forge-admin, please rerender

**Comment by m3vaz** [comment]
This should work once the 11.8 anvil image is available.

**Comment by jakirkham** [comment]
This would simply involving adding a PR to include the new CUDA versions in [conda-forge's docker-images repo]( https://github.com/conda-forge/docker-images ).

For example here's a PR adding CUDA 11.7 ( https://github.com/conda-forge/docker-images/pull/214 ). In particular only [the `.github/workflows/ci.yaml` file]( https://github.com/conda-forge/docker-images/blob/d948295308cc06cc6d147612d4c5bd645522c42c/.github/workflows/ci.yaml ) should need updating (to add the new CUDA 11.8 versions). Would suggest just copying the 11.7 entries and changing to 11.8.

**Comment by m3vaz** [comment]
@conda-forge-admin, please restart ci

**Comment by m3vaz** [comment]
re-running CI since the 11.8 bits are now available. Once this is merged, I will create a new PR for 11.8.1

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,10 +10,10 @@ package:
 
 source:
   url: https://github.com/NVIDIA/cuda-python/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: c8e9d769a6066bb5f94bb0362dbc20ed6f6b525336373263aeb6f9c6dc7b91e4
+  sha256: afc4f0ac46c0e734a71f97d52390424aec1bc9fceb324e30095ca09bc678ff72
 
 build:
-  number: 1
+  number: 0
   skip: true  # [py<37 or osx or cuda_compiler_version != "11.7"]
```
Comment (jakirkham) [inline recipe/meta.yaml:17]: Does this need to be updated?

**Comment by m3vaz** [comment]
I appreciate how you commented on the 11.7 references as I was working on it

**Comment by m3vaz** [comment]
@conda-forge-admin, please rerender

**Comment by m3vaz** [comment]
Clearly most of the headers are being installed. I remember we had to update the windows link/sha in the CI files for 11.8 from the original contents of my PR. It might not be pointing to the right installer. Digging around more today.

**Comment by leofang** [comment]
@m3vaz See https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/216#issuecomment-1340315260.

**Comment by leofang** [comment]
@conda-forge-admin, please restart ci

**Review by leofang — APPROVED** [review]
Windows is green now. Set automerge.

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**